multiplatform-markdown-renderer/src/commonMain/kotlin/com/mikepenz/markdown/model/MarkdownImageState.kt

```diff
@@ -0,0 +1,64 @@
+package com.mikepenz.markdown.model
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.geometry.isUnspecified
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.IntSize
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.unit.toSize
+
+internal interface MarkdownImageState {
+    val imageSize: TextUnitSize
+    fun setContainerSize(intSize: IntSize)
+    fun setImageSize(size: Size)
+}
+
+internal class MarkdownImageStateImpl(private val density: Density) : MarkdownImageState {
+
+    private var parentSize by mutableStateOf(Size.Unspecified)
+
+    private var intrinsicImageSize by mutableStateOf(Size.Unspecified)
+
+    override val imageSize by derivedStateOf {
+        with(density) {
+            if (parentSize.isUnspecified) {
+                TextUnitSize(180.sp, 180.sp)
+            } else if (intrinsicImageSize.isUnspecified) {
+                TextUnitSize(parentSize.width.toSp(), 180.sp)
+            } else {
```

Comment:
Using `TextSize` for an image? Would it maybe make more sense to have this exposed as normal `IntSize` instead? 🤔

Reply:
I made this sp size class because the Placeholder needs sp dimensions and doing conversion here is better. I might be able to remove it. Let's see.